oap-server/server-query-plugin/prometheus-query-plugin/src/main/java/org/apache/skywalking/oap/query/prometheus/handler/PrometheusApiHandler.java

```diff
@@ -0,0 +1,544 @@
+            regex = metric.get();
+        }
+        List<MetricDefinition> definitionList = metricsQuery.listMetrics(regex);
+        int maxNum = definitionList.size();
+        if (limit.isPresent()) {
+            maxNum = limit.get();
+        }
+        for (int i = 0; i < maxNum; i++) {
+            List<MetricMetadata> metadataList = new ArrayList<>();
+            MetricMetadata metadata = new MetricMetadata(MetricType.gauge, "", "");
+            metadataList.add(metadata);
+            response.getData().put(definitionList.get(i).getName(), metadataList);
+        }
+        return jsonResponse(response);
+    }
+
+    @Get
+    @Post
+    @Path("/api/v1/labels")
+    public HttpResponse labels(
+        @Param("match[]") Optional<String> match,
+        @Param("start") Optional<String> start,
+        @Param("end") Optional<String> end) throws IOException {
+        LabelsQueryRsp response = new LabelsQueryRsp();
+        if (match.isPresent()) {
+            PromQLLexer lexer = new PromQLLexer(
+                CharStreams.fromString(match.get()));
+            lexer.addErrorListener(ParseErrorListener.INSTANCE);
+            PromQLParser parser = new PromQLParser(new CommonTokenStream(lexer));
+            parser.addErrorListener(ParseErrorListener.INSTANCE);
+            ParseTree tree;
+            try {
+                tree = parser.expression();
+            } catch (ParseCancellationException e) {
+                response.setStatus(ResultStatus.error);
+                response.setErrorType(ErrorType.bad_data.name());
+                response.setError(e.getMessage());
+                return jsonResponse(response);
+            }
+            PromQLMatchVisitor visitor = new PromQLMatchVisitor();
+            MatcherSetResult parseResult = visitor.visit(tree);
+            String metricName = parseResult.getMetricName();
+            Optional<ValueColumnMetadata.ValueColumn> valueColumn = ValueColumnMetadata.INSTANCE.readValueColumnDefinition(
+                metricName);
+            if (valueColumn.isPresent()) {
+                ValueColumnMetadata.ValueColumn metaData = valueColumn.get();
+                Scope scope = Scope.Finder.valueOf(metaData.getScopeId());
+                Column.ValueDataType dataType = metaData.getDataType();
+                response.getData().addAll(buildLabelNames(scope, dataType));
+            }
+        } else {
+            Arrays.stream(LabelName.values()).forEach(label -> {
+                response.getData().add(label);
+            });
+        }
+        response.setStatus(ResultStatus.success);
+        return jsonResponse(response);
+    }
+
+    @Get
+    @Post
+    @Path("/api/v1/label/{label_name}/values")
+    public HttpResponse labelValues(
+        @Param("label_name") String labelName,
+        @Param("match[]") Optional<String> match,
+        @Param("start") Optional<String> start,
+        @Param("end") Optional<String> end) throws IOException {
```

Comment:
Why are these 3 parameters not used but listed here?

Reply:
Reserve these param to keep consistent with API protocol, have add commons on methods